sdk/storage/azure-storage-blob/azure/storage/blob/_shared/avro/schema.py

```diff
@@ -0,0 +1,1231 @@
+
+            # Validate the fullname:
+            if _RE_FULL_NAME.match(self._fullname) is None:
+                raise SchemaParseException(
+                    'Invalid schema name %r infered from name %r and namespace %r.'
+                    % (self._fullname, self._name, self._namespace))
+
+    def __eq__(self, other):
+        if not isinstance(other, Name):
+            return NotImplemented
+        return self.fullname == other.fullname
+
+    @property
+    def simple_name(self):
+        """Returns: the simple name part of this name."""
+        return self._name
+
+    @property
+    def namespace(self):
+        """Returns: this name's namespace, possible the empty string."""
+        return self._namespace
+
+    @property
+    def fullname(self):
+        """Returns: the full name."""
+        return self._fullname
+
+
+# ------------------------------------------------------------------------------
+
+
+class Names(object):
+    """Tracks Avro named schemas and default namespace during parsing."""
+
+    def __init__(self, default_namespace=None, names=None):
+        """Initializes a new name tracker.
+
+        Args:
+          default_namespace: Optional default namespace.
+          names: Optional initial mapping of known named schemas.
+        """
+        if names is None:
+            names = {}
+        self._names = names
+        self._default_namespace = default_namespace
+
+    @property
+    def names(self):
+        """Returns: the mapping of known named schemas."""
+        return self._names
+
+    @property
+    def default_namespace(self):
+        """Returns: the default namespace, if any, or None."""
+        return self._default_namespace
+
+    def NewWithDefaultNamespace(self, namespace):
+        """Creates a new name tracker from this tracker, but with a new default ns.
+
+        Args:
+          namespace: New default namespace to use.
+        Returns:
+          New name tracker with the specified default namespace.
+        """
+        return Names(names=self._names, default_namespace=namespace)
+
+    def GetName(self, name, namespace=None):
+        """Resolves the Avro name according to this name tracker's state.
+
+        Args:
+          name: Name to resolve (absolute or relative).
+          namespace: Optional explicit namespace.
+        Returns:
+          The specified name, resolved according to this tracker.
+        """
+        if namespace is None:
+            namespace = self._default_namespace
+        return Name(name=name, namespace=namespace)
+
+    def has_name(self, name, namespace=None):
+        avro_name = self.GetName(name=name, namespace=namespace)
+        return avro_name.fullname in self._names
+
+    def get_name(self, name, namespace=None):
+        avro_name = self.GetName(name=name, namespace=namespace)
+        return self._names.get(avro_name.fullname, None)
+
+    def GetSchema(self, name, namespace=None):
```

Comment:
do we want to follow the same naming format? to make all method names in the format like get_schema()?
Ctrl+Shift+r will help you replace the name across files, just be careful not to replace the name in other packages lol

Reply:
done